### PR TITLE
fix validation for AppID length

### PR DESCRIPTION
**Pull request by Syuparn**
# Description

fix validation for AppID length in sidecar injector.

## Issue reference

Please reference the issue this PR will close: #4023

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by msftgits** [comment]
[CLA assistant check](https://cla.opensource.microsoft.com/dapr/dapr?pullRequest=4024) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/dapr/dapr?pullRequest=4024) before we can accept your contribution.<br/><br/>:x: Syuparn [sign now](https://cla.opensource.microsoft.com/dapr/dapr?pullRequest=4024)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/dapr/dapr?pullRequest=4024) it.</sub>

File: pkg/validation/validation_test.go [hunk 1]
```diff
@@ -29,6 +29,16 @@ func TestValidationForKubernetes(t *testing.T) {
 		assert.Error(t, err)
 	})
 
+	t.Run("invalid length if suffix -dapr is appended", func(t *testing.T) {
+		id := ""
```
Comment (daixiang0) [inline pkg/validation/validation_test.go:33]: Just use a const here is enough, no need to use loop.

Comment (Syuparn) [reply pkg/validation/validation_test.go]: I replaced the loop with a single value in 4f5af0f6e40d6dc2b78536a09ad2e864545f0fe5.
`strings.Repeat` is used for length readability.

**Review by mukundansundar — APPROVED** [review]

**Comment by yaron2** [comment]
Thanks for your contribution @Syuparn and welcome to the Dapr community.

**Pull request merged**